crates/engine/tree/src/tree/mod.rs

```diff
@@ -2381,21 +2389,70 @@ where
                         .map_err(|e| InsertBlockErrorKind::Other(Box::new(e)))?,
                 );
 
-                let state_root_task = StateRootTask::new(
-                    state_root_config.clone(),
-                    self.state_root_task_pool.clone(),
-                );
+                let state_root_task =
+                    StateRootTask::new(state_root_config.clone(), self.thread_pool.clone());
+                let state_root_sender = state_root_task.state_root_message_sender();
                 let state_hook = Box::new(state_root_task.state_hook()) as Box<dyn OnStateHook>;
-                (Some(state_root_task.spawn()), Some(state_root_config), state_hook)
+                (
+                    Some(state_root_task.spawn()),
+                    Some(state_root_config),
+                    Some(state_root_sender),
+                    state_hook,
+                )
             } else {
-                (None, None, Box::new(NoopHook::default()) as Box<dyn OnStateHook>)
+                (None, None, None, Box::new(NoopHook::default()) as Box<dyn OnStateHook>)
             };
 
+        // Use cached state provider before executing, used in execution after prewarming threads
+        // complete
+        let caches = ProviderCacheBuilder::default().build_caches();
+        let cache_metrics = CachedStateMetrics::zeroed();
+        let state_provider = CachedStateProvider::new_with_caches(
+            state_provider,
+            caches.clone(),
+            cache_metrics.clone(),
+        );
+
+        if self.config.use_caching_and_prewarming() {
```

Comment:
we could check here `self.config.use_state_root_task()` too, prewarming only makes sense when we use state root task?

Reply:
> we could check here `self.config.use_state_root_task()` too, prewarming only makes sense when we use state root task?

answering myself, nope, i see later that we only calculate multiproof targets and send the prefetch message when we use state root task (ie. state root sender is some), but always execute the transaction